docs/template_plugin/src/template_infer_request.cpp

```diff
@@ -266,19 +283,97 @@ void TemplateInferRequest::waitPipeline() {
 }
 // ! [infer_request:infer_postprocess]
 
+Blob::Ptr TemplateInferRequest::GetBlob(const std::string& name) {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, "GetBlob");
+    InputInfo::Ptr foundInput;
+    DataPtr foundOutput;
+    Blob::Ptr data;
+    const SizeVector oneVector = { 1 };
+    if (findInputAndOutputBlobByName(name, foundInput, foundOutput)) {
+        // ROI blob is returned only if it was set previously. Otherwise default blob is returned.
+        auto it = _preProcData.find(name);
+        if (it != _preProcData.end()) {
+            data = it->second->getRoiBlob();
+        } else {
+            data = _inputs[name];
+            const auto& dims = m_realShapes.find(name) != m_realShapes.end() ? m_realShapes[name] : foundInput->getTensorDesc().getDims();
+            if (data) {
+                if (data->getTensorDesc().getDims() != dims) {
+                    // TODO: implement something smart here instead of raw re-allocation
+                    data.reset();
+                }
+            }
+            if (!data) {
+                auto&& parameters = _executableNetwork->_function->get_parameters();
+                AllocateImplSingle(_inputs, _deviceInputs, *_networkInputs.find(name), [&] (const std::string& blobName) {
+                    return parameters.at(_executableNetwork->_inputIndex.at(blobName))->get_element_type();
+                }, dims);
+                data = _inputs[name];
+            }
+            checkBlob(data, name, true,
+                      foundInput->getTensorDesc().getLayout() != SCALAR
+                      ? dims
+                      : oneVector);
+        }
+    } else {
+        data = _outputs[name];
+        SizeVector dims;
+        if (foundOutput->getTensorDesc().getPartialShape().is_static()) {
+            dims = foundOutput->getTensorDesc().getDims();
+        } else if (_outputTensors[_executableNetwork->_outputIndex.at(name)]->get_partial_shape().is_static()) {
+            dims = _outputTensors[_executableNetwork->_outputIndex.at(name)]->get_shape();
+        } else {
+            IE_THROW() << "Output blob dimensions are not all known for output name " <<
+                name << " with partial shape: " << foundOutput->getTensorDesc().getPartialShape();
+        }
+
+        if (data) {
+            if (data->getTensorDesc().getDims() != dims) {
+                // TODO: implement something smart here instead of raw re-allocation
+                data.reset();
+            }
+        }
+
+        if (!data) {
+            auto &&results = _executableNetwork->_function->get_results();
+            AllocateImplSingle(_outputs, _networkOutputBlobs, *_networkOutputs.find(name),
+                               [&](const std::string &blobName) {
+                                   return results.at(_executableNetwork->_outputIndex.at(blobName))->get_element_type();
+                               }, dims);
+            data = _outputs[name];
+        }
```

Comment:
should not output blob be allocated during port infer stage? While this method just provides empty `Blob::Ptr` to user..